python/ray/train/predictor.py

```diff
@@ -124,6 +124,33 @@ def set_preprocessor(self, preprocessor: Optional[Preprocessor]) -> None:
         """Set the preprocessor to use prior to executing predictions."""
         self._preprocessor = preprocessor
 
+    @classmethod
+    @DeveloperAPI
+    def preferred_batch_format(cls) -> BatchFormat:
+        """Batch format hint for upstream producers to try yielding best block format.
+
+        The preferred batch format to use if both `_predict_pandas` and
+        `_predict_numpy` are implemented. Defaults to Pandas.
+
+        Can be overriden by predictor classes depending on the framework type,
+        e.g. TorchPredictor prefers Numpy and XGBoostPredictor prefers Pandas as
+        native batch format.
+
+        """
+        return BatchFormat.PANDAS
+
+    @classmethod
+    def _batch_format_to_use(cls) -> BatchFormat:
+        """Determine the batch format to use for the predictor."""
+        has_pandas_implemented = cls._predict_pandas != Predictor._predict_pandas
+        has_numpy_implemented = cls._predict_numpy != Predictor._predict_numpy
+        if has_pandas_implemented and has_numpy_implemented:
+            return cls.preferred_batch_format()
+        elif has_pandas_implemented:
+            return BatchFormat.PANDAS
+        else:
+            return BatchFormat.NUMPY
```

Comment:
Nit: The `not has_pandas_immplemented and not has_numpy_implemented` case would fall through here, I'm sure this would be caught elsewhere upstream, but just wanted to call this out. We might want to change this to something like this to be extra defensive:
```suggestion
        elif has_numpy_implemented:
            return BatchFormat.NUMPY
        else:
            raise NotImplementedError("Predictor {cls.__name__} must implement at least one of `_predict_pandas` and `_predict_numpy`")
```